Source/Core/Common/NandPaths.cpp

```diff
@@ -21,41 +21,46 @@ std::string RootUserPath(FromWhichRoot from)
   return File::GetUserPath(idx);
 }
 
-std::string GetImportTitlePath(u64 title_id, FromWhichRoot from)
+static std::string RootUserPath(std::optional<FromWhichRoot> from)
+{
+  return from ? RootUserPath(*from) : "";
+}
+
+std::string GetImportTitlePath(u64 title_id, std::optional<FromWhichRoot> from)
 {
   return RootUserPath(from) + StringFromFormat("/import/%08x/%08x",
                                                static_cast<u32>(title_id >> 32),
                                                static_cast<u32>(title_id));
 }
 
-std::string GetTicketFileName(u64 title_id, FromWhichRoot from)
+std::string GetTicketFileName(u64 title_id, std::optional<FromWhichRoot> from)
 {
   return StringFromFormat("%s/ticket/%08x/%08x.tik", RootUserPath(from).c_str(),
-                          static_cast<u32>(title_id >> 32), static_cast<u32>(title_id));
+                          (u32)(title_id >> 32), (u32)title_id);
```

Comment:
Any reason for dropping the static_cast here?

Reply:
Oops, rebase fail here. I've fixed it now -- thanks for catching this!